model-optimizer/extensions/front/tf/CTCGreedyDecoderReplacement_test.py

```diff
@@ -0,0 +1,159 @@
+                             ('decoder', 'sparse_to_dense', {'out': 0, 'in': 0}),
+                             ('decoder', 'cast', {'out': 1, 'in': 0}),
+                             ('cast', 'sparse_to_dense', {'out': 0}),
+                             ('sparse_to_dense', 'last', {'out': 0, 'in': 0}),
+                             ], nodes_with_edges_only=True)
+        graph.stage = 'front'
+        CTCGreedyDecoderReplacement2().find_and_replace_pattern(graph)
+
+        graph_ref = build_graph(nodes_attributes,
+                                [('logits', 'transpose', {'out': 0}),
+                                 ('transpose', 'shape', {'out': 0}),
+                                 ('transpose', 'shape_1', {'out': 0}),
+                                 ('transpose', 'decoder', {'out': 0, 'in': 0}),
+                                 ('shape', 'strided_slice', {'out': 0, 'in': 0}),
+                                 ('stack', 'strided_slice', {'out': 0, 'in': 1}),
+                                 ('stack1', 'strided_slice', {'out': 0, 'in': 2}),
+                                 ('stack2', 'strided_slice', {'out': 0, 'in': 3}),
+                                 ('shape_1', 'strided_slice_1', {'out': 0, 'in': 0}),
+                                 ('stack_1', 'strided_slice_1', {'out': 0, 'in': 1}),
+                                 ('stack1_1', 'strided_slice_1', {'out': 0, 'in': 2}),
+                                 ('stack2_1', 'strided_slice_1', {'out': 0, 'in': 3}),
+                                 ('strided_slice', 'unsqueeze_batch_size', {'out': 0, 'in': 0}),
+                                 ('unsqueeze_batch_size_axis', 'unsqueeze_batch_size', {'out': 0, 'in': 1}),
+                                 ('strided_slice_1', 'unsqueeze_time_size', {'out': 0, 'in': 0}),
+                                 ('unsqueeze_time_size_axis', 'unsqueeze_time_size', {'out': 0, 'in': 1}),
+                                 ('unsqueeze_batch_size', 'seq_mask_shape', {'out': 0, 'in': 1}),
+                                 ('unsqueeze_time_size', 'seq_mask_shape', {'out': 0, 'in': 0}),
+                                 ('one', 'sequence_mask', {'out': 0, 'in': 0}),
+                                 ('seq_mask_shape', 'sequence_mask', {'out': 0, 'in': 1}),
+                                 ('sequence_mask', 'decoder', {'out': 0, 'in': 1}),
+                                 ('decoder', 'squeeze_dec_seq', {'out': 0, 'in': 0}),
+                                 ('squeeze_axes', 'squeeze_dec_seq', {'out': 0, 'in': 1}),
+                                 ('squeeze_dec_seq', 'cast_to_int', {'out': 0, 'in': 0}),
+                                 ('cast_to_int', 'last', {'out': 0, 'in': 0}),
+                                 ],
+                                nodes_with_edges_only=True)
+
+        (flag, resp) = compare_graphs(graph, graph_ref, 'last', check_op_attrs=True)
+        self.assertTrue(flag, resp)
```

Comment:
you care about node names in the transformations -- please check it in the test

Reply:
Fixed